src/FluentMigrator.Runner/MigrationRunner.cs

```diff
@@ -169,17 +137,28 @@ public void MigrateUp(long targetVersion)
         public void MigrateUp(long targetVersion, bool useAutomaticTransactionManagement)
         {
             var migrationInfos = GetUpMigrationsToApply(targetVersion);
+
+            if (!migrationInfos.Any()) return;
```

Comment:
Please apply `.ToList()` to `GetUpMigrationsToApply` and use `.Count` to avoide reenumeration.